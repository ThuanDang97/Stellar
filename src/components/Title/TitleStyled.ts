import styled from 'styled-components'

// Themes
import { theme } from '@themes/index'

// Utils
import { pxToRem } from '@utils/theme'

// Props type
import { TextProps } from '@self-types/components/Text.props'

// Components
import { TextStyled } from '../styled-components/TextStyled'

export const TitleStyled = styled(TextStyled)<TextProps>`
  color: ${(props) => props.color || theme.colors.white};
  font-size: ${(props) =>
    (props.fontSize && pxToRem(props.fontSize)) ||
    pxToRem(theme.typography.fontSize.xlg)};
  text-align: ${(props) => props.textAlign || 'center'};
  font-weight: ${(props) =>
    props.fontWeight || theme.typography.fontWeight.bold};
  line-height: ${(props) =>
    (props.lineHeight && pxToRem(props.lineHeight)) ||
    pxToRem(theme.typography.lineHeight.xl)};
  margin: ${(props) => (props.margin && pxToRem(props.margin)) || 0};
  @media (max-width: ${pxToRem(theme.metrics.breakPoints.sm)}) {
    font-size: ${pxToRem(theme.typography.fontSize.default)};
    line-height: ${pxToRem(theme.typography.lineHeight.xmd)};
  }
`

export const TitlePageStyled = styled(TitleStyled)<TextProps>`
  @media (max-width: ${pxToRem(theme.metrics.breakPoints.sm)}) {
    font-size: ${pxToRem(theme.typography.fontSize.default)};
    line-height: ${pxToRem(theme.typography.lineHeight.xmd)};
    padding-top: ${pxToRem(theme.metrics.dimensions.md)};
  }
`

export const TitlePopupStyled = styled(TitleStyled)<TextProps>`
  @media (max-width: ${pxToRem(theme.metrics.breakPoints.xs)}) {
    font-size: ${pxToRem(theme.typography.fontSize.md)};
    line-height: ${pxToRem(theme.typography.lineHeight.md)};
  }
`
