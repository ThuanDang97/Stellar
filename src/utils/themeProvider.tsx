import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
