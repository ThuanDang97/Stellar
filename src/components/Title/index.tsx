import React, { memo } from 'react'

// Components
import { TitleStyled } from '@components/Title/TitleStyled'

// Props type
import { TitleProps } from '@self-types/components/Title.props'

const Title = ({
  color,
  fontSize,
  fontWeight,
  title,
  lineHeight,
  headingLevel = 'h1',
  titleHover,
  ...props
}: TitleProps) => (
  <TitleStyled
    as={headingLevel}
    color={color}
    fontSize={fontSize}
    fontWeight={fontWeight}
    lineHeight={lineHeight}
    title={titleHover}
    {...props}
  >
    {title}
  </TitleStyled>
)

export default memo(Title)
