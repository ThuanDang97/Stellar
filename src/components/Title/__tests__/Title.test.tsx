// Components
import Title from '@components/Title/index'

// Themes
import { theme } from '@themes/index'

// Utils
import { renderWithTheme } from '@utils/themeProvider'

describe('Title Component', () => {
  const props = {
    title: 'Start Paying Bills & Building Credit With Stellar',
    color: theme.colors.black,
    size: theme.typography.fontSize.xlg,
    weight: theme.typography.fontWeight.bold,
    align: 'center',
  }

  test('Component Title matches DOM Snapshot', () => {
    const { asFragment } = renderWithTheme(<Title {...props} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
