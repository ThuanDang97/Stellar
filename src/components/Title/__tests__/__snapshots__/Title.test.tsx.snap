// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component Component Title matches DOM Snapshot 1`] = `
<DocumentFragment>
  <h1
    class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 bfcxuE WstJK"
    color="#000"
    size="36"
  >
    Start Paying Bills & Building Credit With Stellar
  </h1>
</DocumentFragment>
`;
